pkg/logs/internal/framer/framer.go

```diff
@@ -124,6 +124,17 @@ func (fr *Framer) GetFrameCount() int64 {
 // Process handles an incoming chunk of data.  It will call outputFn for any recognized frames.  Partial
 // frames are maintained between calls to Process.  The passed buffer is not used after return.
 func (fr *Framer) Process(input *message.Message) {
+	// we can only process unstructured message in the framer
+	// TODO(remy): the same way the MultiLineHandler use the first part
+	// of a structured message to recompose partials ones into only one,
+	// we might consider doing the same on structured log messages with
+	// the framer.
+	if input.State != message.StateUnstructured {
```

Comment:
Is this check needed?
Framers are configured per tailer (hard coded). Both journald and windows use `framer.NoFraming` which (I think?) should effectivly do the same thing as this condiitonal check. 

I can't think of a scenario where we would be putting strucutred messages in front of a framer.

Reply:
You are right, it's more an "early return" than anything else since the `NoFraming` thingy will still run all the lines of code below, [copying memory around](https://github.com/DataDog/datadog-agent/pull/19968/files/3e98ec03e93c9c0d0608d7614210dbb07732e89f#diff-566194ce3bbce552094ead4fe4f24d6cf338734af0f1f45a9cf11f52a21291beR176-R178) but with no actual impact whatsoever on the content...